pymc/sampling/jax.py

```diff
@@ -175,25 +175,29 @@ def _sample_stats_to_xarray(posterior):
     return data
 
 
+def _device_put(input, device: str):
+    return jax.device_put(input, jax.devices(device)[0])
+
+
 def _postprocess_samples(
-    jax_fn: List[TensorVariable],
+    jax_fn: Callable,
     raw_mcmc_samples: List[TensorVariable],
-    postprocessing_backend: str,
-    num_chunks: Optional[int] = None,
+    postprocessing_backend: Literal["cpu", "gpu"] | None = None,
+    postprocessing_vectorize: Literal["vmap", "scan"] | None = None,
 ) -> List[TensorVariable]:
-    if num_chunks is not None:
-        loop = xmap(
-            jax_fn,
-            in_axes=["chain", "samples", ...],
-            out_axes=["chain", "samples", ...],
-            axis_resources={"samples": SerialLoop(num_chunks)},
+    if postprocessing_vectorize is None or postprocessing_vectorize == "scan":
+        t_raw_mcmc_samples = [jnp.swapaxes(t, 0, 1) for t in raw_mcmc_samples]
```

Comment:
Is the xmap approach broken? I would consider that to be better than using scan

Reply:
yes, xmap with `SerialLoop(num_chunks)` is broken and the bug seems to be model dependent so instead of that it is possible to loop over with scan

Reply:
Could you share a bit more how it is broken?

Reply:
I see, thanks for the context. What happens if you set `postprocessing_chunks=1`?
I am fine with the change, only minor concern here is that using scan will be slower, and also we are deprecating a kwarg directly which we should have a deprecation circle

Reply:
``postprocessing_chunks=1`` still fails

Reply:
Well, we tried 🤷

Reply:
Does using scan instead of xmap result in large memory usage again. (ie: undo-ing this solution https://discourse.pymc.io/t/out-of-memory-when-transforming-variables-in-numpyro-jax/9904/9 ?)

Reply:
Using Scan shouldn't be worse memory wise, vmap was the problem IIRC. However that `lax.map` helper seems a bit more readable than using scan ourselves (it does so for that). CC @ferrine